AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -126,14 +126,25 @@ - (AVPlayerItem *)constructPlayerItem
 {
   ASDN::MutexLocker l(__instanceLock__);
 
+  AVPlayerItem *playerItem = nil;
   if (_asset != nil) {
-    AVPlayerItem *playerItem = [[AVPlayerItem alloc] initWithAsset:_asset];
-    playerItem.videoComposition = _videoComposition;
-    playerItem.audioMix = _audioMix;
-    return playerItem;
+    if ([_asset isKindOfClass:[AVURLAsset class]] && [self hasURLAsset]) {
+      playerItem = [[AVPlayerItem alloc] initWithURL:((AVURLAsset *)_asset).URL];
+      _asset = [playerItem asset];
+    } else {
+      playerItem = [[AVPlayerItem alloc] initWithAsset:_asset];
+    }
   }
 
-  return nil;
+  playerItem.videoComposition = _videoComposition;
+  playerItem.audioMix = _audioMix;
+  return playerItem;
+}
+
+- (BOOL)hasURLAsset
+{
+  // The array of AVAssetTrack objects available via the tracks property of an URL asset is typically empty for streaming-based media
+  return _asset.tracks.count == 0;
```

Comment:
@Xinchi Below, it looks like there is use of the safe / "correct" API to asynchronously load this value.  Can we instead use that form of request / completion handler to restructure how we construct the player item, so the player item is created "as quickly as possible" even if it first goes through a call to loadValues...completionBlock: and the bulk of the work is in the completion block?